Source/player.cpp

```diff
@@ -3146,91 +3146,81 @@ BOOL __fastcall PM_DoSpell(int pnum)
 // 52571C: using guessed type int drawpanflag;
 // 5BB1ED: using guessed type char leveltype;
 
-int __fastcall PM_DoGotHit(int pnum)
+BOOL __fastcall PM_DoGotHit(int pnum)
 {
-	int v1; // esi
-	int v2; // eax
-	int v3; // edx
-	int v4; // ecx
-
-	v1 = pnum;
-	if ( (unsigned int)pnum >= MAX_PLRS )
+	if ( (DWORD)pnum >= MAX_PLRS ) {
 		TermMsg("PM_DoGotHit: illegal player %d", pnum);
-	v2 = v1;
-	v3 = plr[v1]._pIFlags;
-	v4 = plr[v1]._pAnimFrame;
-	if ( v3 & 0x200000 && v4 == 3 )
-		plr[v2]._pAnimFrame = 4;
-	if ( v3 & 0x400000 && (v4 == 3 || v4 == 5) )
-		++plr[v2]._pAnimFrame;
-	if ( v3 & 0x800000 && (v4 == 1 || v4 == 3 || v4 == 5) )
-		++plr[v2]._pAnimFrame;
-	if ( plr[v2]._pAnimFrame < plr[v2]._pHFrames )
-		return 0;
-	StartStand(v1, plr[v2]._pdir);
-	ClearPlrPVars(v1);
-	if ( random(3, 4) )
-		ArmorDur(v1);
-	return 1;
+	}
+
+	int frame = plr[pnum]._pAnimFrame;
+	if ( plr[pnum]._pIFlags & ISPL_FASTRECOVER && frame == 3 ) {
+		plr[pnum]._pAnimFrame = 4;
+	}
+	if ( plr[pnum]._pIFlags & ISPL_FASTERRECOVER && (frame == 3 || frame == 5) ) {
+		plr[pnum]._pAnimFrame++;
+	}
+	if ( plr[pnum]._pIFlags & ISPL_FASTESTRECOVER && (frame == 1 || frame == 3 || frame == 5) ) {
+		plr[pnum]._pAnimFrame++;
+	}
+
+	if ( plr[pnum]._pAnimFrame < plr[pnum]._pHFrames ) {
+		return FALSE;
+	}
+
+	StartStand(pnum, plr[pnum]._pdir);
+	ClearPlrPVars(pnum);
+	if ( random(3, 4) ) {
+		ArmorDur(pnum);
+	}
+
+	return TRUE;
 }
 
 void __fastcall ArmorDur(int pnum)
 {
-	int v1; // ebp
-	//int v2; // ST04_4
-	PlayerStruct *v3; // esi
-	int v4; // eax
-	int v5; // edi
-	int v6; // esi
-	int v7; // ecx
-	int v8; // ecx
-	unsigned char v9; // dl
+	if ( pnum != myplr ) { //5

```

Comment:
What are the `//5`, `//8`, `//19`, etc comments? These should probably be removed.

Reply:
It's the correlating line numbers from the original ASM; yeah I forgot to remove them when things where fixed :) Done now.